exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/xsort/ExternalSortBatch.java

```diff
@@ -223,7 +227,18 @@ public void close() {
         if (mSorter != null) {
           mSorter.clear();
         }
-
+        for(Iterator iter = this.currSpillDirs.iterator(); iter.hasNext(); iter.remove()) {
+            Path path = (Path)iter.next();
+            try {
+                if (fs != null && path != null && fs.exists(path)) {
+                    if (fs.delete(path, true)) {
+                        fs.cancelDeleteOnExit(path);
+                    }
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
Done. An exception throw was replaced with:
```
// since this is meant to be used in a batches's cleanup, we don't propagate the exception
final String message = "Unable to delete spill directory \"" + path + "\"";
System.err.println(message + '.');
logger.warn(message, e);
```